src/app/model/ChessEngine.ts

```diff
@@ -68,7 +70,78 @@ export class ChessEngine implements ChessModel {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
 
+  private returnWirtualProposedBoardAfterMove(chessboard: Chessboard, squareFrom: Square, squareTo: Square): Chessboard {
+    const wirtualPorposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = wirtualPorposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    return wirtualPorposedChessboard;
+  }
+
+  public static kingPosition(chessboard: Chessboard, kingSide: Side): Square | undefined {
+    const { squaresWithPiece: squaresWithPieces } = chessboard;
+    const kingPositionKey = Object.keys(squaresWithPieces).find((key) => {
+      const isKingName = squaresWithPieces[key].name === 'King';
+      const isPlayerSide = squaresWithPieces[key].side === kingSide;
+      return isKingName && isPlayerSide;
+    });
+    const kingPosition = kingPositionKey
+      ? {
+          column: kingPositionKey[0],
+          row: Number(kingPositionKey[1]) as Row,
+        }
+      : undefined;
+
+    return kingPosition;
+  }
+
+  public static isSquareChecked(chessboard: Chessboard, playerSide: Side, positionToControl: Square): boolean {
+    let isCheckedSquareFlag = false;
+    const { squaresWithPiece: squaresWithPieces } = chessboard;
+
+    Object.keys(squaresWithPieces).forEach((key) => {
+      const mappedPiece = squaresWithPieces[key];
+      const isEnemySide = mappedPiece.side !== playerSide;
+
+      if (isEnemySide) {
+        const mappedPiecePosition = {
+          column: key[0],
+          row: Number(key[1]) as Row,
+        };
+        const possibleMappedPieceMoves = mappedPiece.possibleMoves(mappedPiecePosition, chessboard);
+        const isKingPositionOnPossibleEnemyPieceMoves = possibleMappedPieceMoves.some(
+          (checkedPosition) => JSON.stringify(checkedPosition) == JSON.stringify(positionToControl),
+        );
+        if (isKingPositionOnPossibleEnemyPieceMoves) isCheckedSquareFlag = true;
+      }
+    });
+    return isCheckedSquareFlag;
+  }
+
+  public static isKingChecked(chessboard: Chessboard, kingSide: Side): boolean {
+    const kingPosition = this.kingPosition(chessboard, kingSide);
+    const isKingChecked = kingPosition ? this.isSquareChecked(chessboard, kingSide, kingPosition) : false;
+    return isKingChecked;
+  }
```

Comment:
To moznaby skrocic:
```suggestion
  public static isKingChecked(chessboard: Chessboard, kingSide: Side): boolean {
    const kingPosition = this.kingPosition(chessboard, kingSide);
    return this.isSquareChecked(chessboard, kingSide, kingPosition);
  }
```

Reply:
Właśnie nie wiem kiedy ta ostatnia linijka z return zawiera za dużo znaków.  Wszak tu ładujemy już 3 argumenty. Czy to nie jest sporo? Nie wspominałeś, że jeśli tworzymy takie długie to lepiej wsadzić do zmiennej? 
(Początkowo tak zrobiłem, ale zmieniłem)
@nowakprojects